app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -87,6 +89,13 @@ def render_invalid_shop_error
       redirect_to return_address
     end
 
+    def enable_cookie_access
+      fullpage_redirect_to enable_cookies_path(
+        shop: sanitized_shop_name,
+        return_to: params[:return_to]
```

Comment:
Should this be `session[:return_to]`?

Reply:
@ragalie - This is correct because at this point in the execution, the user has been redirected before Safari allows the session to be set, so `session[:return_to]` is blank here, even though it was set within `LoginProtection`.

Reply:
Isn't it set here though? https://github.com/Shopify/shopify_app/pull/664/files#diff-350e7187d4b8bbcd1eaee3ff44fc7fb3R48

I'd rather us handle `params[:return_to]` in a limited number of places and use `session[:return_to]` as the source of truth for this value except where we do the params processing.